react/src/base/inputs/SprkDatePicker/SprkDatePicker.js

```diff
@@ -0,0 +1,184 @@
+import React, { Component } from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+import TinyDatePicker from 'tiny-date-picker';
+import assign from 'lodash/assign';
+import uniqueId from 'lodash/uniqueId';
+
+class SprkDatePicker extends Component {
+  constructor(props) {
+    const { value } = props;
+    super(props).datePickerInputRef = React.createRef();
+    this.tdpConfig = {
+      mode: 'dp-below',
+      lang: {
+        days: ['S', 'M', 'T', 'W', 'T', 'F', 'S'],
+      },
+      min: '01/1/2008',
+      max: '01/1/2068',
+      format: (date) =>
+        date
+          .toLocaleDateString('en-US', {
+            month: '2-digit',
+            day: '2-digit',
+            year: 'numeric',
+          })
+          .replace(/[^ -~]/g, ''),
+    };
+
+    if (value) {
+      this.state = {
+        hasValue: true,
+      };
+    } else {
+      this.state = {
+        hasValue: false,
+      };
+    }
+  }
+
+  componentDidMount() {
+    const { datePickerConfig } = this.props;
+    TinyDatePicker(
+      this.datePickerInputRef.current,
+      assign(this.tdpConfig, datePickerConfig),
+    );
+  }
+
+  render() {
+    const {
+      datePickerConfig,
+      variant,
+      additionalClasses,
+      analyticsString,
+      formatter,
+      idString,
+      isDisabled,
+      isValid,
+      ariaDescribedBy,
+      value,
+      id,
+      ...rest
+    } = this.props;
+    const { hasValue } = this.state;
+
+    // Adds class for IE and Edge
+    const handleOnBlur = (e) => {
+      if (e.target.value.length > 0) {
+        this.setState({ hasValue: true });
+      } else {
+        this.setState({ hasValue: false });
+      }
+    };
+
+    return (
+      <input
+        className={classNames('sprk-b-TextInput', additionalClasses, {
+          'sprk-b-TextInput--error': !isValid,
+          'sprk-b-Input--has-floating-label': hasValue && variant === 'huge',
+        })}
+        disabled={isDisabled}
+        ref={this.datePickerInputRef}
+        aria-describedby={ariaDescribedBy}
+        data-id={idString}
+        data-analytics={analyticsString}
+        aria-invalid={!isValid}
+        value={isValid && formatter ? formatter(value) : value}
```

Comment:
//TODO, only run formatter when value has changed, as is, this gets run everytime.